fhd_core/gridding/baseline_grid_locations.pro

```diff
@@ -0,0 +1,158 @@
+    flag_dist_baseline=where((dist_test_min LT min_baseline) $
+      OR (dist_test_max GT max_baseline),n_dist_flag)
+  ENDIF
+  dist_test=0
+  dist_test_min=0
+  dist_test_max=0
+
+  ; Create the other half of the uv plane via negating the locations
+  conj_i=where(ky_arr GT 0,n_conj)
+  IF n_conj GT 0 THEN BEGIN
+    kx_arr[conj_i]=-kx_arr[conj_i]
+    ky_arr[conj_i]=-ky_arr[conj_i]
+  ENDIF
+
+  ; Center of baselines for x and y in units of pixels
+  xcen=Float(frequency_array#Temporary(kx_arr))
+  ycen=Float(frequency_array#Temporary(ky_arr))
+
+  ; Pixel number offet per baseline for each uv-box subset 
+  x_offset=Fix(Floor((xcen-Floor(xcen))*psf_resolution) mod psf_resolution, type=12) ; type=12 is unsigned int
+  y_offset=Fix(Floor((ycen-Floor(ycen))*psf_resolution) mod psf_resolution, type=12) ; type=12 is unsigned int
+  
+  IF keyword_set(interp_flag) THEN BEGIN
+    ; Derivatives from pixel edge to baseline center for use in interpolation
+    dx_arr = (xcen-Floor(xcen))*psf_resolution - Floor((xcen-Floor(xcen))*psf_resolution)
+    dy_arr = (ycen-Floor(ycen))*psf_resolution - Floor((ycen-Floor(ycen))*psf_resolution)
+    dx0dy0_arr = (1-dx_arr)*(1-dy_arr)
+    dx0dy1_arr = (1-dx_arr)*dy_arr
+    dx1dy0_arr = dx_arr*(1-dy_arr)
+    dx1dy1_arr = Temporary(dx_arr) * Temporary(dy_arr)
+  ENDIF
+
+  ; The minimum pixel in the uv-grid (bottom left of the kernel) that each baseline contributes to
+  xmin=Long(Floor(Temporary(xcen))+dimension/2-(psf_dim/2-1))
+  ymin=Long(Floor(Temporary(ycen))+elements/2-(psf_dim/2-1))
+
+  ; Set the minimum pixel value of baselines which fall outside of the uv-grid to -1 to exclude them
+  range_test_x_i=where((xmin LE 0) OR ((xmin+psf_dim-1) GE dimension-1),n_test_x)
+  range_test_y_i=where((ymin LE 0) OR ((ymin+psf_dim-1) GE elements-1),n_test_y)
+  IF n_test_x GT 0 THEN xmin[range_test_x_i]=(ymin[range_test_x_i]=-1)
+  IF n_test_y GT 0 THEN xmin[range_test_y_i]=(ymin[range_test_y_i]=-1)
+  
+  ; Flag baselines which fall outside the uv plane
+  IF ~keyword_set(fill_model_visibilities) THEN BEGIN
+    IF n_dist_flag GT 0 THEN BEGIN
+      ; If baselines fall outside the desired min/max baseline range at all during the frequency range, 
+      ; then set their minimum pixel value to -1 to exlude them 
+      xmin[*,flag_dist_baseline]=-1
+      ymin[*,flag_dist_baseline]=-1
+      flag_dist_baseline=0
+    ENDIF
+  ENDIF 
+
+  IF vis_weight_switch THEN BEGIN
+    ; If baselines are flagged via the weights, then set their minimum pixel value to -1 to exclude them
+    flag_i=where(vis_weights LE 0,n_flag)
+    IF Keyword_Set(fill_model_visibilities) THEN n_flag=0L
+    IF n_flag GT 0 THEN BEGIN
+      xmin[flag_i]=-1
+      ymin[flag_i]=-1
+    ENDIF
+    vis_weights=0
+    flag_i=0
+  ENDIF
+
+  IF Keyword_Set(mask_mirror_indices) THEN BEGIN
+    ; Option to exlude v-axis mirrored baselines
+    IF n_conj GT 0 THEN BEGIN
+      xmin[*,conj_i]=-1
+      ymin[*,conj_i]=-1
+    ENDIF
+  ENDIF
+
+END
```

Comment:
You can also include the following code here:
`; Match all visibilities that map from and to exactly the same pixels and store them as a histogram in bin_n
; with their respective index ri. Setting min equal to 0 excludes flagged (i.e. (xmin,ymin)=(-1,-1)) data
bin_n=histogram(xmin+ymin*dimension,binsize=1,reverse_indices=ri,min=0)`

I think it might make sense to turn `baseline_grid_locations` into a function and return `bin_n`, but I don't feel strongly about that.